ocaml/xapi/xapi_vm_group_helpers.ml

```diff
@@ -40,24 +40,27 @@ let check_breach_on_vm_anti_affinity_rules ~__context ~group =
         Some h
 
 let report_anti_affinity_alert ~__context ~group ~host =
-  let group_uuid = Db.VM_group.get_uuid ~__context ~self:group in
-  let host_uuid = Db.Host.get_uuid ~__context ~self:host in
-  let body =
-    String.concat ""
-      [
-        "<body><message>Breach on VM anti-affinity rules</message><VM_group>"
-      ; group_uuid
-      ; "</VM_group><host>"
-      ; host_uuid
-      ; "</host></body>"
-      ]
-  in
-  let obj_uuid =
-    Db.Pool.get_uuid ~__context ~self:(Helpers.get_pool ~__context)
-  in
-  Xapi_alert.add
-    ~msg:Api_messages.all_running_vms_in_anti_affinity_grp_on_single_host
-    ~cls:`Pool ~obj_uuid ~body
+  if Pool_features.is_enabled ~__context Features.VM_anti_affinity then
+    let group_uuid = Db.VM_group.get_uuid ~__context ~self:group in
+    let host_uuid = Db.Host.get_uuid ~__context ~self:host in
+    let body =
+      String.concat ""
+        [
```

Comment:
It's probably not easy to add: but how rules are violated would be useful to know.

Reply:
The breaching for anti-affinity rule can be accessed in this PR: https://github.com/xapi-project/xen-api/pull/5605